nav2_smoother/test/test_simple_smoother.cpp

```diff
@@ -276,5 +272,5 @@ TEST(SmootherTest, test_simple_smoother)
   max_its_path.poses[9].pose.position.y = 0.9;
   max_its_path.poses[10].pose.position.x = 0.5;
   max_its_path.poses[10].pose.position.y = 1.0;
-  EXPECT_FALSE(smoother->smooth(max_its_path, max_time));
+  EXPECT_THROW(smoother->smooth(max_its_path, max_time), nav2_core::FailedToSmoothPath);
```

Comment:
Adding the failed to smooth exception exposes that this test currently doesn't do anything. 

The path segments are less than 9 thus nothing gets smoothed. 
https://github.com/ros-planning/navigation2/blob/main/nav2_smoother/src/simple_smoother.cpp#L74

The reason it returns true is because success is initialized to true.